### ISSUE TITLE
txt2tags 2.7?

### ISSUE THREAD

**Issue by cypriani** [txt2tags/txt2tags#207]
Hi,
The current stable version of txt2tags is 6 years old, and the development version fixes several bugs. Is a new stable release planned any time soon?
Matteo

**Comment by farvardin** [comment]
I don't think it's planned yet :(
But I think it's a necessity too. The new improvements really should be in a new 2.7 version!

**Comment by aureliojargas** [comment]
First step: Test-suite should run 100% green.

Second step: Make sure the test suite won't break again (use Pull Requests, respect Travis CI).

**Comment by d0ugm** [comment]
Dear Sir:

I get below error.
I am using python 3.5.3 on windows.

  File "txt2tags.py", line 2122
    print '\033[32;1m%s\033[m'%msg
                             ^
SyntaxError: Missing parentheses in call to 'print'

**Comment by fgallaire** [comment]
@d0ugm txt2tags only supports Python2

**Comment by selkfoster** [comment]
Thanks to the developers of txt2tags to keep the project alive! -- Please, publish a newer version of it, even if it is not ready for Python3.  Thank you!

**Comment by Petrichor31** [comment]
With Python 2.7 obsolescence in 2020, txt2tags future seems uncertain.
Maybe, it will be interesting to split the project in 2:
- txt2tags specification defining all the syntax. So anyone could code a software on this basis
- the software converting text to HTML and other.
As well, I miss a Visual Studio Code plugin.

**Comment by brdjns** [comment]
Are there any plans for a conversion to Python 3.5.x?

**Comment by farvardin** [comment]
I agree with the previous message (from september 2018), we should split
the project in 2.

About a conversion to Python 3.5, I don't know about a plan for porting it.
I can't do it myself, and I think txt2tags should be ported to something
more useful than python (javascript?).

I see there is this project for supporting only one code between python 2
and python 3: https://pypi.org/project/six/

I've also tried to use the script 2to3 which can convert python2 code to
python3, but I get some errors while using txt2tags with python3 so the
conversion is not perfect...



Le ven. 1 févr. 2019 à 12:11, Bradley Jones <notifications@github.com> a
écrit :

**Comment by jendrikseipp** [comment]
I ported the old txt2tags 2.6 to Python 3 a while ago: https://github.com/jendrikseipp/rednotebook/blob/master/rednotebook/external/txt2tags.py

(I chose to port the released version instead of the latest revision from the repo because the released version was more stable and I didn't need any features added in later revisions.) 

It contains the following disclaimer:

```
# This file is based on txt2tags version 2.6, but has been modified for
# RedNotebook. The changes compared to the upstream version are:
#
#   * use spaces instead of tabs
#   * port to Python 3
#   * don't escape underscores in tagged and raw LaTeX text
#   * don't use locale-dependent str.capitalize()
#   * support SVG images
```

Maybe it is useful for people who prefer a Python 3 version.

**Comment by farvardin** [comment]
Why waiting too much? I've had a look at the test errors, we have only this:

csv: 1 error for lout (I don't this this one is very popular among txt2tags users)
db: several errors for aat and txt (not used that much in my opinion)

It's time to release 2.7 and prepare the python 3 release as well.

**Comment by farvardin** [comment]
For lout, it seems the "blockVerbOpen" tag expected in the test was replaced by "tableOpen" during the CSV convertion. I don't know if it was on purpose and if it's legit in the Lout format.

**Comment by fgallaire** [comment]
@farvardin I don't understand in which way JavaScript is _more useful_ than Python
and _releasing_ txt2tags is not a simple word, it's a work only @aureliojargas can properly do

**Comment by lbartoletti** [comment]
> I ported the old txt2tags 2.6 to Python 3 a while ago: https://github.com/jendrikseipp/rednotebook/blob/master/rednotebook/external/txt2tags.py
> 
> (I chose to port the released version instead of the latest revision from the repo because the released version was more stable and I didn't need any features added in later revisions.)
> 
> It contains the following disclaimer:
> 
> ```
> # This file is based on txt2tags version 2.6, but has been modified for
> # RedNotebook. The changes compared to the upstream version are:
> #
> #   * use spaces instead of tabs
> #   * port to Python 3
> #   * don't escape underscores in tagged and raw LaTeX text
> #   * don't use locale-dependent str.capitalize()
> #   * support SVG images
> ```
> 
> Maybe it is useful for people who prefer a Python 3 version.

Can you create a dedicated repository for it please?

**Comment by farvardin** [comment]
hello,

Several useful features have been added to the development version of txt2tags (especially the template feature). But they were probably not needed by rednotebook.

It's also the reason why a "release often, release early" policy is a better solution, if we had done this the development state of txt2tags would be better, and not stalled like now.

**Comment by jendrikseipp** [comment]
> Can you create a dedicated repository for it please?

I created a dedicated repository for txt2tags version 2.6 ported to Python 3 at https://github.com/jendrikseipp/txt2tags.

**Comment by farvardin** [comment]
Good job. I'll have a look at your repository. I've removed myself from the current organisation as it leads to nowhere.

**Comment by junghans** [comment]
> > Can you create a dedicated repository for it please?
> 
> I created a dedicated repository for txt2tags version 2.6 ported to Python 3 at https://github.com/jendrikseipp/txt2tags.

We are now using this version in Fedora, too:
https://src.fedoraproject.org/rpms/txt2tags/c/3c1d15d98c8ef78b58f3177b13894b565850336a?branch=master

**Comment by aureliojargas** [comment]
Since Dec 17, 2019 Jendrik's fork https://github.com/jendrikseipp/txt2tags has been merged back into the main txt2tags repository, under the `v3` branch https://github.com/txt2tags/txt2tags/tree/v3

It is now the official branch for txt2tags version 3.x.

Current version 2.7-dev has been moved to the `v2` branch https://github.com/txt2tags/txt2tags/tree/v2.

Read about the differences from 2x and 3.x series in the `main` branch README https://github.com/txt2tags/txt2tags/blob/master/README.md
